--- a/sysdeps/ieee754/flt-32/s_finitef.c
+++ b/sysdeps/ieee754/flt-32/s_finitef.c
@@ -22,20 +22,24 @@
  * no branching!
  */
 
-#include <math.h>
-#include <math_private.h>
-
-#undef __finitef
-
-#ifndef FINITEF
-# define FINITEF __finitef
-#endif
-
-int FINITEF(float x)
-{
-	int32_t ix;
-	GET_FLOAT_WORD(ix,x);
-	return (int)((u_int32_t)((ix&0x7fffffff)-0x7f800000)>>31);
-}
-hidden_def (__finitef)
-weak_alias (__finitef, finitef)
+/* FINITEF and __finitef ALREADY DEFINED INLINE IN math_private.h; SKIP THIS FILE
+ * 
+ * #include <math.h>
+ * #include <math_private.h>
+ * 
+ * #undef __finitef
+ * 
+ * #ifndef FINITEF
+ * # define FINITEF __finitef
+ * #endif
+ * 
+ * int FINITEF(float x)
+ * {
+ * 	int32_t ix;
+ * 	GET_FLOAT_WORD(ix,x);
+ * 	return (int)((u_int32_t)((ix&0x7fffffff)-0x7f800000)>>31);
+ * }
+ * hidden_def (__finitef)
+ * weak_alias (__finitef, finitef)
+ * 
+ */
